### PR TITLE
Modorders 158 refactor po line model schemas

### PR DESCRIPTION
**Pull request by pautau-s**
## Purpose
[MODORDERS-158](https://issues.folio.org/browse/MODORDERS-158) Refactor POLine model - Schemas

## Approach
The poLine now includes: adjustments, claims, cost, detail, eresource, fund_distribution, license, location, physical, source, vendor_detail
-  id and poLineId fields are removed from the sub-object schemas listed above
-  UUID references to these sub-objects are removed from PoLine and their schemas are included/referenced by the poLine schema instead. (a la compositePoLine)
-  examples are updated
-  fields schemas are transitioned to camelCase.

## Pre-Merge Checklist
Before merging this PR, please go through the following list and take appropriate actions.

- Does this PR meet or exceed the expected quality standards?
  - [x] Examples exist for all schemas
  - [x] Descriptions exist for all schema properties
  - [x] All schemas pass raml-cop linting
- Does this introduce breaking changes?
  - [x] Were there any schema changes?
  - [ ] There are no breaking changes in this PR.
  
If there are breaking changes, please **STOP** and consider the following:

- What other modules will these changes impact?
- Do JIRAs exist to update the impacted modules?
  - [ ] If not, please create them
  - [ ] Do they contain the appropriate level of detail?  Which endpoints/schemas changed, etc.
  - [ ] Do they have all they appropriate links to blocked/related issues?
- Are the JIRAs under active development?  
  - [ ] If not, contact the project's PO and make sure they're aware of the urgency.
- Do PRs exist for these changes?
  - [ ] If so, have they been approved?

Ideally all of the PRs involved in breaking changes would be merged in the same day to avoid breaking the folio-testing environment.  Communication is paramount if that is to be achieved, especially as the number of intermodule and inter-team dependencies increase.  

While it's helpful for reviewers to help identify potential problems, ensuring that it's safe to merge is ultimately the responsibility of the PR assignee.

### CODE REVIEW

File: mod-orders-storage/examples/alert_collection.sample [hunk 1]
```diff
@@ -4,8 +4,8 @@
           {
             "alert": "Receipt overdue",
             "id": "9a665b22-9fe5-4c95-b4ee-837a5433c95d",
-            "po_line_id": "8c778aee-97fa-4586-b131-3ea588a728e2"
+            "poLineId": "8c778aee-97fa-4586-b131-3ea588a728e2"
```
Comment (grabsefx) [inline mod-orders-storage/examples/alert_collection.sample]: Schema `alert.json` do not have `poLineId` field anymore. So it should be deleted from the sample
```suggestion
```

Comment (grabsefx) [inline mod-orders-storage/examples/alert_collection.sample]: Schema `alert.json` do not have `poLineId` field anymore. So it should be deleted from the sample

**Comment by piotr-kalashuk** [comment]
> Should the .json model files also be deleted along with the .sample files for models that have been flattened into the poLine record?

My personal preference is to keep references to the existing schema files as instead of including content of "sub-object" schemas. At least this will simplify support of the `acq-models/composite_po_line.json` and `acq-models/mod-orders-storage/schemas/po_line.json` schemas which are going to be almost the same except of reporting codes, alerts etc.

File: mod-orders-storage/examples/alert_collection.sample (hunk 1 above)
Comment (pautau-s) [reply mod-orders-storage/examples/alert_collection.sample]: Fixed. Thank.

File: mod-orders-storage/schemas/cost.json [hunk 2]
```diff
@@ -3,39 +3,29 @@
...
       "description": "a currency code",
       "type": "string"
     },
-    "quantity_physical": {
+    "quantityPhysical": {
       "description": "quantity of physical items in this purchase order line",
       "type": "integer"
     },
-    "quantity_electronic": {
+    "quantityPlectronic": {
```
[the remark this answers is not in the document]
Comment (pautau-s) [reply mod-orders-storage/schemas/cost.json]: Thanks. Done.

**Comment by craigmcnally** [comment]
As discussed with the team, let's merge this now but hold off on merging the module code

**Pull request merged**